### PR TITLE
Add Tekton task for generating provenance 

### PR DESCRIPTION
**Pull request by priyawadhwa**
Adds a task which:
1. pulls in the distroless repo at the specified commit
2. builds the images and pushes to an insecure registry to the cluster
3. gets the digests for six distroless images, and saves them in the IMAGES result

This six images will be picked up by Tekton Chains to generate provenance

If this works, we can definitely add in more images. we can probably have up to 30 per Task before we hit the maximum size for Tekton results.

### CODE REVIEW

**Review by dlorenc — APPROVED** [review]

File: cloudbuild.yaml [hunk 1]
```diff
@@ -60,3 +60,23 @@ steps:
...
+- name: gcr.io/cloud-builders/gcloud
+  entrypoint: sh
+  args:
+  - -c
+  - |
+    #!/bin/sh
+    set -o errexit
+    set -o xtrace
+
+    gcloud container clusters get-credentials provenance --zone=us-central1-c --project=distroless
```
Comment (mattmoor) [inline cloudbuild.yaml:74]: Use `${PROJECT_ID}` here?

File: cloudbuild.yaml [hunk 2]
```diff
@@ -60,3 +60,23 @@ steps:
...
+    
+    # Install tkn
+    curl -Lo tkn_0.20.0_Linux_arm64.tar.gz  https://github.com/tektoncd/cli/releases/download/v0.20.0/tkn_0.20.0_Linux_x86_64.tar.gz
+    tar -xvzf tkn_0.20.0_Linux_arm64.tar.gz -C /usr/local/bin/ tkn
```
Comment (mattmoor) [inline cloudbuild.yaml:78]: Hoist a version variable for 0.20.0 to make bumping this easier?

File: provenance/provenance-task.yaml [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        value: insecure-registry.default.svc.cluster.local:5000
+      - name: COMMIT_SHA
+        value: $(params.CHAINS-GIT_COMMIT)
+      script: |
+        #!/bin/sh
+        set -ex
+
+        git clone $(params.CHAINS-GIT_URL)
+        cd distroless
+        git checkout $(params.CHAINS-GIT_COMMIT)
```
Comment (mattmoor) [inline provenance/provenance-task.yaml:37]: It might be good to make this another step?

Comment (priyawadhwa) [reply provenance/provenance-task.yaml:37]: agreed! i'm going to convert this to a pipeline in another PR, so i'll add it in there.

**Pull request merged**